templates/shared/header-nav.html

```diff
@@ -1,14 +1,13 @@
-{% macro link(path_link, page_path, display_text) %}
-<a {% if page_current is startingwith page_path -%} class="active" {% endif -%}
-  href="{{path_link}}">{{display_text}}</a>
+{% macro link(page_path, display_text) %}
+<a {% if page_current==page_path -%} class="active" {% endif -%} href="{{ fmturl(page_path) }}">{{ display_text }}</a>
```

Comment:
I kind of want to keep the current `page_current` matching logic.

Reply:
This isn't a blocking objection to the change as it doesn't affect the main functionality for this PR but isn't the current `page_current` matching logic redundant? What is the benefit of not matching `page_current` with the value used in `href` and having `page_link` and `page_path` be different?

Reply:
It's admittedly semantics that don't matter for this project, but

1. Since the highlighting is now tied to the templates rather than the routes defined in axum, I felt like it made sense to just separate it completely.
2. The separation and matching on prefix theoretically allows for nested navigation styling, though that's not likely to be needed here.